yt/yt/server/job_proxy/environment.cpp

```diff
@@ -954,6 +1016,39 @@ class TCriJobProxyEnvironment
 
 private:
     const TCriJobEnvironmentConfigPtr Config_;
+
+    NCGroups::TSelfCGroupsStatisticsFetcher StatisticsFetcher_;
+
+    std::optional<TJobEnvironmentMemoryStatistics> DoGetJobMemoryStatistics() const
+    {
+        auto statistics = StatisticsFetcher_.GetMemoryStatistics();
+        // NB: PeakRss usage is intentional here, to make statistics more sane.
+        return TJobEnvironmentMemoryStatistics{
+            .Rss = statistics.PeakRss,
```

Comment:
Maybe call it as PeakRss in this case?

Reply:
I'd prefer to do it in a separate PR with some additional statistics refactorings.